src/main/java/com/employeesService/EmployeesService/model/Department.java

```diff
@@ -36,11 +37,17 @@ public void setDeptName(String deptName) {
         this.deptName = deptName;
     }
 
-    public void addEmployee(DepartmentEmployee departmentEmployee) {
-        departmentEmployees.add(departmentEmployee);
+    public void addEmployee(Employee employee, LocalDate fromDate) {
+        DepartmentEmployee employeeToAdd = new DepartmentEmployee();
+        DepartmentEmployeeId id = new DepartmentEmployeeId(this.deptNo, employee.getEmpNo());
+        employeeToAdd.setId(id);
+        employeeToAdd.setDepartment(this);
+        employeeToAdd.setEmployee(employee);
```

Comment:
Isn't it easier to do like: `new DepartmentEmployee(this, employee)` and then set the `fromDate` 

All the stuff you're doing here, is done inside that constructor, aren't they?

Reply:
Yes you're right, for the moment i have a constructor that accepts a Department and an Employee, should i change it to accept also the Id or should i let it like this with the setId(id) method

Reply:
That's enough, as id is assigned in the constructor, isn't it?

Reply:
Im creating the Id with "DepartmentId id=new DepartmentId(this.deptNo, employee.getEmpNo());" and then im assigning that as the id of "employeeToAdd with the setter method `employeeToAdd.setId(id);` , should i leave it like this or should i make it so it get assigned in the constructor?

Reply:
![image](https://user-images.githubusercontent.com/11609385/141573954-e980f69f-9efb-46f7-b647-6ea37626f7f0.png)

Reply:
Ouu...right